metaflow/runtime.py

```diff
@@ -910,17 +910,22 @@ def get_args(self):
         # TODO: Make one with dict_to_cli_options; see cli_args.py for more detail
         def _options(mapping):
             for k, v in mapping.items():
-                if v:
-                    # we need special handling for 'with' since it is a reserved
-                    # keyword in Python, so we call it 'decospecs' in click args
-                    if k == "decospecs":
-                        k = "with"
-                    k = k.replace("_", "-")
-                    v = v if isinstance(v, (list, tuple, set)) else [v]
-                    for value in v:
-                        yield "--%s" % k
-                        if not isinstance(value, bool):
-                            yield to_unicode(value)
+
+                # None or False arguments are ignored
+                # v needs to be explicitly False, not falsy, eg. 0 is an acceptable value
+                if v is None or v is False:
+                    continue
+
+                # we need special handling for 'with' since it is a reserved
+                # keyword in Python, so we call it 'decospecs' in click args
+                if k == "decospecs":
+                    k = "with"
```

Comment:
Separately, there is actually no need for this clause here but this whole thing needs to be reworked with dict_to_cli_options in cli_args.py. For consistency, could you change that one as well?

Reply:
I'm not sure I understand. Should we be unifying everything to use the `dict_to_cli_options` function in `utils.py`?

Reply:
Longer term yes, for now, there is a function that looks almost identical in `cli_args.py` and I was suggesting to make the same change there.